worlds/banjo_tooie/Hints.py

```diff
@@ -0,0 +1,578 @@
+
+        locationName.JIGGYGM2,
+        locationName.JIGGYGM5,
+
+        locationName.JIGGYWW1,
+        locationName.JIGGYWW5,
+
+        locationName.JIGGYJR1,
+        locationName.JIGGYJR3,
+
+        locationName.JIGGYTD2,
+        locationName.JIGGYTD9,
+
+        locationName.JIGGYGI6,
+        locationName.JIGGYGI9,
+
+        locationName.JIGGYHP3,
+        locationName.JIGGYHP6,
+        locationName.JIGGYHP8,
+        locationName.JIGGYHP10,
+
+        locationName.JIGGYCC4,
+
+        locationName.SCRUT,
+        locationName.SCRAT,
+    ])
+
+    if world.options.randomize_jinjos:
+        bad_location_names.extend([
+            locationName.JIGGYIH5,
+            locationName.JIGGYIH6,
+
+            locationName.JINJOGM1,
+            locationName.JINJOCC1,
+        ])
+
+    if world.options.randomize_cheato:
+        bad_location_names.extend([
+            locationName.CHEATOCC2,
+            locationName.CHEATOTL1,
+            locationName.CHEATOGM1,
+        ])
+
+    # The next 5 most expensive silos
+    if world.options.randomize_bt_moves:
+        sorted_silos = [k for k, v in sorted(world.jamjars_siloname_costs.items(), key=lambda item: item[1])]
+        for _ in range(6, 10):
+            bad_location_names.append(sorted_silos.pop())
+
+    if world.options.cheato_rewards:
+        bad_location_names.extend([
+            locationName.CHEATOR3,
+        ])
+
+    if world.options.honeyb_rewards:
+        bad_location_names.extend([
+            locationName.HONEYBR4,
+        ])
+    return bad_location_names
+
+
+def generate_move_hints(world: "BanjoTooieWorld", hints: List[Hint]):
+    move_locations = get_move_locations(world)
+    for location in move_locations:
+        hints.append(Hint(world, location))
+
+
+def get_move_locations(world: "BanjoTooieWorld") -> List[Location]:
+    all_moves_names = []
+
+    # We don't want BT moves to be hinted when they're in the vanilla location.
+    if world.options.randomize_bt_moves:
+        all_moves_names.extend(moves_table.keys())
+    all_moves_names.extend(bk_moves_table.keys())
+    all_moves_names.extend(progressive_ability_table.keys())
+
+    all_move_locations = [location for location in get_all_hintable_locations(world)
+                          if location.item.name in all_moves_names and location.item.player == world.player]
+    world.random.shuffle(all_move_locations)
+    selected_move_locations = []
+
+    for location in all_move_locations:
+        if len(selected_move_locations) >= min(
+            world.options.signpost_move_hints.value,
+            world.options.signpost_hints.value
+        ):
+            return selected_move_locations
+        selected_move_locations.append(location)
+    return selected_move_locations
+
+
+def get_location_by_name(world: "BanjoTooieWorld", name: str) -> Location | None:
+    potential_match = list(filter(lambda location: location.name == name, get_player_hintable_locations(world)))
```

Comment:
```suggestion
    potential_match = [location for location in get_player_hintable_locations(world) if location.name == name]
```
Don't see why not to use a comprehension for this one.